slim.Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM python:3.7-alpine as builder
+
+WORKDIR /girder
+RUN apk add --no-cache git gcc musl-dev libffi-dev openssl-dev
+COPY . ./
+# Both PYTHONDONTWRITEBYTECODE and --no-compile are necessary to avoid creating .pyc files
+RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir --no-compile .
```

Comment:
A curious property in how docker composes layers is that even if a file is unchanged between the builder and runtime images, it will still be copied and take up space on two image layers.  As such, when we do `COPY --from=builder /usr/local/lib/python3.7/site-packages/ /usr/local/lib/python3.7/site-packages/`, the preinstalled site-packages end up using space twice.  This could be avoided by doing something like:

```
RUN touch /tmp/timestamp

RUN PYTHONDONTWRITEBYTECODE=1 pip install --no-cache-dir --no-compile .

RUN find /usr/local/lib/python3.7/site-packages/ -type f -a -not -newer /tmp/timestamp -delete
```

which obviously breaks the builder's image (since we just deleted pip and some other modules), but will reduce the size of the copy layer by about 6 Mbytes.

Reply:
On the fence about whether or not this is worth it.  Someone else should weigh in.